File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/repository/Repository.java
package com.fixture.shop.repository;

import java.util.List;

import com.fixture.shop.models.Order;
import com.fixture.shop.models.OrderStatus;
import com.fixture.shop.models.Product;

/**
 * Storage contract. goToImplementation on Repository should resolve to
 * InMemoryRepository and PostgresRepository.
 */
public interface Repository {

    void saveOrder(Order order);

    Order getOrder(String orderId);

    void updateOrderStatus(String orderId, OrderStatus status);

    Product getProduct(String sku);

    List<Order> listOrdersByUser(String userId);
}

File: skill_sdk/tests/fixtures/ts-project/src/shop/repository/Repository.ts
import { Order } from '../models/Order';
import { OrderStatus } from '../models/OrderStatus';
import { Product } from '../models/Product';

/**
 * Storage contract. goToImplementation on Repository should resolve to
 * InMemoryRepository and PostgresRepository.
 */
export interface Repository {
  saveOrder(order: Order): void;
  getOrder(orderId: string): Order | undefined;
  updateOrderStatus(orderId: string, status: OrderStatus): void;
  getProduct(sku: string): Product | undefined;
  listOrdersByUser(userId: string): Order[];
}
